### PR TITLE
Fix case exports order columns for deprecated groups

### PR DESCRIPTION
**Pull request by mkangia**
## Product Description

Minor follow up to https://github.com/dimagi/commcare-hq/pull/34502 to correct ordering on case export columns based on data dictionary for deprecated case property group.

This only effects projects with access to Data Dictionary.

## Technical Summary

https://dimagi.atlassian.net/browse/SC-3674

Some context,
So when a case property group is deprecated, the case properties in that group are removed from that group by setting their group to be blank. In the UI, the properties are not moved to the "No Group" section until there is a page reload after save.
In such a case when data dictionary is saved, we were retaining the index of case properties as it was on the group that was deprecated. An "index" stores the index of the case property in its group to maintain the order of case properties.
This would cause "index" on these case properties to be faulty until next save was done, which would then correctly set the index.

So, this PR changes the index to the default value of 0 for case properties belonging to case property group getting deprecated on save.
Additionally, to make ordering of case properties with the same index, additionally ordering by their primary key, to make their order predictable and same for both case exports and data dictionary. It would not be very rare for two case properties to have the same index like set in the case above or for case properties that getting automatically added to Data dictionary. Only a save on Data Dictionary would set the index on the case properties.

### Safety story

Tested locally. Blast radius is limited to Data dictionary view or new case exports for projects with access to data dictionary. The additional indexing with pk is not expected to cause any performance issues.

### QA Plan

https://dimagi.atlassian.net/browse/QA-6489

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js [hunk 1]
```diff
@@ -235,7 +235,7 @@ hqDefine("data_dictionary/js/data_dictionary", [
                             'caseType': element.caseType,
                             'name': element.name,
                             'label': element.label() || element.name,
-                            'index': index,
+                            'index': group.toBeDeprecated() ? 0 : index,
```
Comment (zandre-eng) [inline corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js:238]: If we deprecate the group and then undeprecate it again before saving, would all the case properties be saved with an index of 0?

**Review by zandre-eng — COMMENTED** [review]
Looks good, just one comment from my end.

File: corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js (hunk 1 above)
Comment (mkangia) [reply corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js:238]: No, this is only when saved is called.

Comment (zandre-eng) [reply corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js:238]: Thanks for clarifying, that makes sense.

**Review by zandre-eng — APPROVED** [review]
Looks good from my end :rocket:

**Comment by mkangia** [comment]
Thank you for the quick review @zandre-eng

**Pull request merged**